cabby/geo/walk.py

```diff
@@ -440,11 +447,63 @@ def get_points_and_route(map: map_structure.Map) -> Optional[item.RVSPath]:
                                                             main_pivot,
                                                             near_pivot,
                                                             beyond_pivot,
-                                                            cardinal_direction)
-
+                                                            cardinal_direction,
+                                                            number_intersections)
     return rvs_path_entity
 
 
+def get_number_interesctions_past(main_pivot: Dict, route: GeoDataFrame, map: map_structure.Map, end_point: Point) -> int:
+    '''Return the number of intersections between the main_pivot and goal. If the main_pivot and goal are on different streets return -1.
+    Arguments:
+      main_pivot: The pivot along the route.
+      route: The route along which a landmark will be chosen.
+      map: The map of a specific region.
+      end_point: The goal location.
+    Returns:
+      The number of intersections between the main_pivot and goal. If the main_pivot and goal are on different streets return -1.
+    '''
+    dist = route['geometry'].apply(
+        lambda x: x.distance(main_pivot['geometry']))
+    pivot_idx = dist.idxmin().tolist()
+    pivot_goal_route = route.loc[pivot_idx:]
+
+    if pivot_goal_route.shape[0] <= 1:
+        return -1
+
+    # Check if main pivot is in the pivot_goal_route segment.
+    segment_pivot = LineString(pivot_goal_route['geometry'].iloc[0:2].values)
+    number_intersection = - \
```

Comment:
Can you add a comment here explaining what is going on? Not clear why you are using negative value here? (I'm not sure what project_point_in_segment is doing is this context.) Giving a specific example would help. (Also for the segment_goal part below too.)